### PR TITLE
feat(oracle): Add new DatedExchangeRate query with block timestamp and block height

### PR DESCRIPTION
**Pull request by matthiasmatt**
# Purpose / Abstract

- This implement query required for https://github.com/NibiruChain/nibiru/issues/2075. This PR enables us to give settlement time information for an oracle exchange rate

## Summary by CodeRabbit

- **New Features**
	- Added support for fetching the dated exchange rate from the Nibiru oracle module.
	- Introduced a new RPC method `DatedExchangeRate` for querying the latest price, timestamp, and block height.
	- Enhanced the `queryExchangeRate` function to return more detailed information, including price, block time, and block height.
	- Improved the changelog for better readability and organization.

- **Bug Fixes**
	- Improved clarity and consistency in the changelog entries related to EVM features and fixes.

- **Tests**
	- Added a new test for the `DatedExchangeRate` method to ensure accurate retrieval of exchange rates and proper error handling.

### CODE REVIEW

File: x/oracle/keeper/keeper.go [hunk 1]
```diff
@@ -187,6 +187,16 @@ func (k Keeper) GetExchangeRate(ctx sdk.Context, pair asset.Pair) (price sdk.Dec
 	return
 }
 
+func (k Keeper) GetDatedExchangeRate(ctx sdk.Context, pair asset.Pair) (price sdk.Dec, blockTimeMs int64, BlockHeight uint64, err error) {
+	exchangeRate, err := k.ExchangeRates.Get(ctx, pair)
+	if err != nil {
+		return
+	}
+	time := ctx.WithBlockHeight(int64(exchangeRate.CreatedBlock)).BlockTime()
```
[the remark this answers is not in the document]
Comment (Unique-Divine) [reply x/oracle/keeper/keeper.go:195]: It's not possible for the created block to be in the future

**Review by Unique-Divine — COMMENTED** [review]
LGTM, I'll just add a small rename to the proto message and then merge this.

**Comment by matthiasmatt** [comment]
> LGTM, I'll just add a small rename to the proto message and then merge this.

What would you use? @Unique-Divine

**Review by Unique-Divine — APPROVED** [review]

**Comment by Unique-Divine** [comment]
> > LGTM, I'll just add a small rename to the proto message and then merge this.
> 
> What would you use? @Unique-Divine

Did `height -> block_height` and `timestamp_ms -> block_timestamp_ms`

**Pull request merged**